addons/website/static/tests/tours/website_no_dirty_page.js

```diff
@@ -0,0 +1,45 @@
+/** @odoo-module **/
+
+import tour from 'web_tour.tour';
+import wTourUtils from 'website.tour_utils';
+
+tour.register('website_no_dirty_page', {
+    test: true,
+    url: '/',
+}, [
+    ...wTourUtils.clickOnEditAndWaitEditMode(),
+    wTourUtils.dragNDrop({
+        id: "s_text_image",
+        name: "Text - Image",
+    }), {
+        content: "Click on Discard",
+        trigger: '.o_we_website_top_actions [data-action="cancel"]',
+    }, {
+        content: "Check that discarding actually warns when there are dirty changes, and cancel",
+        trigger: ".modal-footer .btn-secondary",
+    },
+    ...wTourUtils.clickOnSave(),
+    ...wTourUtils.clickOnEditAndWaitEditMode(),
+    {
+        content: "Confirm we are in edit mode",
+        trigger: 'body.editor_has_snippets',
+        run: () => null,
+    }, {
```

Comment:
This is already part of the `clickOnEditAndWaitEditMode` method (ensuring the edit mode is correctly rdy

Reply:
I know but this checks for the `.editor_has_snippets` class explicitely as otherwise checking for `body:not(.editor_has_snippets)` at the final step after discarding would not be testing anything. It would be the same as testing `body:not(.some_random_thing)` which would always be true.